src/App.js

```diff
@@ -3,18 +3,84 @@ import Card from './Card';
 import { useState } from 'react';
 
 function App() {
-  const [checked, setChecked] = useState(false);
+  const [cardState, setCardState] = useState({
+    checked: false,
+    isEditable: false
+  });
+
+  const [cardText, setCardText] = useState({
+    headerText: 'Caption',
+    tempHeader: 'Caption',
+    bodyText: 'text...',
+    tempBody: 'text...'
+  });
```

Comment:
Far more logical would be to move _headerText_ and _bodyText_ to a separate object `data` and put into **cardState** itself (however it's more like app state data rather then card state, just a bit confusing naming).
Meanwhile _tempHeader_ and _tempBody_ should be moved to a separate object `tempData `and put into **Card component** state - it's logically should be placed there. Just think of it - if this will be covered in the Card itself - there should be no changes to this logic once you start operating not a single card but the cards array.
Since you should not update app state each time you fill a character into a field.
Also, if you use the same structure of actual card data and temp card data, you'll be able to use spread operator to put updates for the whole object.